testcases/kernel/syscalls/timer_settime/timer_settime03.c

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) HCL Technologies Ltd,.  All Rights Reserved.
+ *
+ * Author:	Prachi Parekh <prachi.r@hcl.com>
+ *
+ */
+/*
+ * This tests the timer_settime(2) syscall under various conditions:
+ *
+ * 1) Using a periodic timer invoking signal handler
+ * 2) Using absolute timer invoking signal handler
+ *
+ * All of these tests are supposed to be successful.
+ */
+
+#include <stdlib.h>
+#include <errno.h>
+#include <time.h>
+#include <signal.h>
+#include "tst_test.h"
+#include "lapi/common_timers.h"
+#include <stdio.h>
+#include "tst_safe_macros.h"
+
+#define SIG SIGALRM
+
+static struct timespec timenow;
+static struct itimerspec new_set, old_set;
+static kernel_timer_t timer;
+static int handler_var = 0;
```

Comment:
This should be declared volatile as well in order to avoid miscompilation. Anything that is changed in signal handler happens asynchronously to the program and the compiler has no chance to know unless we describe it so.

Reply:
Are you talking about only the below variable "handler_var" ?
```bash
static int handler_var = 0;
```
As this is the only variable changed in signal handler.

Or for all the variables shown above 
```bash
static struct timespec timenow;
static struct itimerspec new_set, old_set;
static kernel_timer_t timer;
static int handler_var = 0;
```

Reply:
Only about the variable changed in the signal handler. The signal handler executes asynchronously to the rest of the code, hence the compiler has no idea to know that the value may change out of the blue.

Reply:
Variables like `handler_var` (which are changed in signal handlers) need to be marked volatile, lest the value might be incorrectly read/stored in a register: https://stackoverflow.com/a/42183083 .